include/uncertainty_planning_core/uncertainty_contact_planning.hpp

```diff
@@ -141,6 +140,100 @@ class UncertaintyPolicyExecutionResult
   }
 };
 
+#if UNCERTAINTY_PLANNING_CORE__SUPPORTED_ROS_VERSION == 2
+struct ExecutionTimeLimit {
```

Comment:
I am not a fan of `structs` generally, especially in a case like this where we can use it to enforce some invariants
```
#if UNCERTAINTY_PLANNING_CORE__SUPPORTED_ROS_VERSION == 2
class ExecutionTimeLimit
{
public:
  explicit ExecutionTimeLimit(const double seconds)
      : seconds_(seconds), clock_(std::make_shared<rclcpp::Clock>())
  {
    if (seconds_ < 0.0)
    {
      throw std::invalid_argument("Time limit cannot be less than zero seconds");
    }
  }

  ExecutionTimeLimit(const double seconds, const rclcpp::Clock::SharedPtr& clock)
      : seconds_(seconds), clock_(clock)
  {
    if (seconds_ < 0.0)
    {
      throw std::invalid_argument("Time limit cannot be less than zero seconds");
    }
    if (!clock_)
    {
      throw std::invalid_argument("No valid clock supplied");
    }
  }

  double Seconds() const { return seconds_; }

  rclcpp::Clock& Clock() { return *clock_; }

private:
  double seconds_ = 0.0;
  rclcpp::Clock::SharedPtr clock_;
};
#elif UNCERTAINTY_PLANNING_CORE__SUPPORTED_ROS_VERSION == 1
class ExecutionTimeLimit
{
public:
  explicit ExecutionTimeLimit(const double seconds) : seconds_(seconds)
  {
    if (seconds_ < 0.0)
    {
      throw std::invalid_argument("Time limit cannot be less than zero seconds");
    }
  }

  double Seconds() const { return seconds_; }

private:
  double seconds_ = 0.0;
};
#endif

Reply:
See https://github.com/calderpg/uncertainty_planning_core/commit/2d3a145375854e0caadd7204655a6f32761e023c.